julia/src/api_types.jl

```diff
@@ -33,33 +33,40 @@ function bust_union(x::TaggedUnion{U}) where U
 end
 bust_union(x) = x  # not a union, leave it alone
 
+"Forces reinterpretting `raw` as a `T`. If sizeof(raw) <: size(T) high bits will be filled with junk."
```

Comment:
Is this even safe if `sizeof(raw) < size(T)`? In that case wouldn't `unsafe_load` read memory out of bounds, potentially leading to a segfault?

Reply:
I *think* we are safe, as we know the memory that is out of bounds is still memory we have access to.
Because C unions take up the space of the largest element.
So we know our program has that memory reserved, so no segfault.
It is probably unsafe in general, but since we just use it for busting up unions, I think we are in the clear.